src/bfv_ske.rs

```diff
@@ -57,38 +58,57 @@ mod tests {
         const Q: u64 = 32;
 
         let (bfv, sk) = Bfv::<N, Q, T>::keygen();
+        let mut rng = rand::rng();
 
-        let m_a_1 = E::new(1);
-        let m_a_2 = E::new(0);
-        let m_a_3 = E::new(1);
-        let m_a_4 = E::new(0);
-        let m_a = Polynomial::<N, T>::new([m_a_1, m_a_2, m_a_3, m_a_4]);
+        let m_a_coeffs: [E; N] = core::array::from_fn(|_| E::new(rng.random_range(0..T) as i64));
+        let m_a = Polynomial::<N, T>::new(m_a_coeffs);
```

Comment:
```suggestion
        let m_a = Polynomial::<N, T>::rand();
```

Also same for all other `random_range` codes